presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcClient.java

```diff
@@ -139,4 +140,9 @@ PreparedStatement getPreparedStatement(Connection connection, String sql)
     String quoted(RemoteTableName remoteTableName);
 
     Map<String, Object> getTableProperties(JdbcIdentity identity, JdbcTableHandle tableHandle);
+
+    default Optional<TableScanRedirectApplicationResult> getTableScanRedirection(ConnectorSession session, SchemaTableName tableName, Set<JdbcColumnHandle> sourceColumns)
```

Comment:
let's remove:
1. `Set<JdbcColumnHandle> sourceColumns` from this method. 
2. pass `JdbcTableHandle tableHandle` instead.